### ISSUE TITLE
[feature proposal] improve access rules management on site

### ISSUE THREAD

**Issue by Gallenkamp** [corona-warn-app/cwa-documentation#817]
Dear devs,

I'd like to propose a solution to the situation on site when waiters, shopkeepers and entrance staff have to discuss and explain new entrance rules to their customers. Rules may change and customers (as well as shopkeepers) are losing track of them.

### Situation
1. Client has either CWA or CovPass app with their personal **status of immunisation**. That is a set of infections and vaxination in any possible combination. So the customers app **knows** if the customer did have "a third shot within the last 90 but not the last 14 days".
2. The shopkeeper knows what **kind of location** he is receiving costomers in. If it's a café, a clothing store, a grocery store. He knows the rules for his shop and has to explain it to customers every time the rules change. Plus: the shop next door might seem similar but have different rules due to legal reasons.

### New features

1. CWA and CovPass app can check in to locations and by doing that also read the local rules from a shop-baser QR code.
2. The customers app can read those access rules 
3. The customers app can cross check with its own data of immunisation status
4. The customers app can diplay the local access rules in readable text
5. The customers app can show  green lights / red lights
6. The customers app can check the missing immunisation action and display additional info like "you need another shot to get access this location" / "you need a test result not older than 48 hous to access this location"

### Pros and cons

- **Pros**
-- Customers can check for themselves and don't have to trust a stranger on rules
-- Conflicts move from person-to-person to person-to-his-own-phone, releasing social stress from shopkeepers etc
-- People don't have to learn the rules for each kind of shop
-- 
- **Cons**
-- Does not solve the shopkeepers obligation to check clients permission
-- Feedback to client can be misleading if shopkeeper has wrong ruleset in QR code
-- ruleset needs to be maintained 
-- Shopkeepers need to have a QR code creation kit for their location + situation

**Comment by Jo-Achim** [comment]
Is with...
>     1. CWA and CovPass app can check in to locations and by doing that also read the local rules from a shop-baser QR code.
... something meant like possible under [event registration](https://www.coronawarn.app/de/eventregistration/)? Only extended to relevant data (access rules)?

If so, I would want to expand a point under "**Cons**":
> -- Shopkeepers need to have a QR code creation kit for their location + situation

   -- Shopkeepers must have the opportunity to check regularly, if not daily, that the rules contained in the check-in-QR-code are actually still valid. If not or in case of doubt, he needs to use the "QR code creation kit for their location + situation" daily to create a new QR-Code.

Right?

**Comment by Gallenkamp** [comment]
> ... something meant like possible under event registration? Only extended to relevant data (access rules)?

Right. With access rules integrated into the checkin QR code, users can check for themselves, even from outside the shop when a QR code is presented in the window.


> -- Shopkeepers must have the opportunity to check regularly, if not daily, that the rules contained in the check-in-QR-code are actually still valid. If not or in case of doubt, he needs to use the "QR code creation kit for their location + situation" daily to create a new QR-Code.

Not really. They have to keep track of the rules anyway. The only additional step is to print out a new QR code once the new rules become effective. That can be done upfront if the QR code generator can handle "current" and "from jan 25th" rulesets.

**Comment by Jo-Achim** [comment]
> > -- Shopkeepers must have the opportunity to check regularly, if not daily, that the rules contained in the check-in-QR-code are actually still valid. If not or in case of doubt, he needs to use the "QR code creation kit for their location + situation" daily to create a new QR-Code.
> 
> Not really. They have to keep track of the rules anyway. The only additional step is to print out a new QR code once the new rules become effective. That can be done upfront if the QR code generator can handle "current" and "from jan 25th" rulesets.

In other words, the tool "QR code creation kit for their location + situation" is a kind of automation that monitors rule changes and - whatever - informs the shopkeeper and tells him that he has to post a new QR code by date "from jan 25th"?
